### PR TITLE
New CAL test key for CI

### PR DESCRIPTION
**Pull request by apaillier-ledger**
## Description

With #327, a private key was set as repository secret to be able to generate signature on-the-fly to test the CAL APDUs.
Being a secret meant that PRs from forks could not access it and Ragger tests would fail.

A new private key, that is only used for the CI tests and isn't even compiled in the production app, has been added to the repository, and now PRs coming from external developers will have functioning Ragger tests.

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [x] Tests
- [ ] Documentation
- [ ] Other (for changes that might not fit in any category)

### CODE REVIEW

File: src/tokens.h [hunk 1]
```diff
@@ -59,6 +66,7 @@ static const uint8_t LEDGER_SIGNATURE_PUBLIC_KEY[] = {
     0x80, 0x13, 0x73, 0x07, 0xde, 0x4d, 0xd1, 0xd4, 0x18, 0x42, 0x8c, 0xf2, 0x1a, 0x93,
     0xb3, 0x35, 0x61, 0xbb, 0x09, 0xd8, 0x8f, 0xe5, 0x79
 #endif
```
Comment (fbeutin-ledger) [inline src/tokens.h:68]: see above to avoid the nesting

File: Makefile [hunk 2]
```diff
@@ -151,6 +151,9 @@ endif
 CAL_TESTING_KEY:=0
 ifneq ($(CAL_TESTING_KEY),0)
 DEFINES += HAVE_CAL_TESTING_KEY
+ifneq ($(CAL_TESTING_KEY),1)
+DEFINES += HAVE_CAL_TESTING_KEY_CI
+endif
 endif
```
Comment (fbeutin-ledger) [inline Makefile:157]: Plus clair je trouve:
```
CAL_TESTING_KEY:=0
CAL_TESTING_KEY_CI:=0
ifneq ($(CAL_TESTING_KEY),0)
DEFINES += HAVE_CAL_TESTING_KEY
endif
ifneq ($(CAL_TESTING_KEY_CI),0)
DEFINES += HAVE_CAL_TESTING_KEY_CI
endif
```

Dans le code .c
#if defined (HAVE_CAL_TESTING_KEY) && defined(HAVE_CAL_TESTING_KEY_CI)
#error "CAL KEY contradiction"
#endif

File: tests/ragger/conftest.py [hunk 3]
```diff
@@ -36,6 +39,13 @@ def firmware(arg_model: str) -> Firmware:
             return fw
     raise ValueError("Unknown device model \"%s\"" % (arg_model))
 
+@pytest.fixture
+def cal_key() -> Optional[SigningKey]:
```
Comment (fbeutin-ledger) [inline tests/ragger/conftest.py:43]: Mmmh not really a fixture as it's not really a "parameter" of the test
I would suggest making a class out of it, also I suggest taking a look into the SigningAuthority defined in the exchange app tests, I instantiate and manipulate the same kind of structure. Maybe unify both and put it in ragger utils ?

File: src/tokens.h [hunk 4]
```diff
@@ -51,6 +51,13 @@ static const uint8_t LEDGER_SIGNATURE_PUBLIC_KEY[] = {
     0xfe, 0x4e, 0x8c, 0xc7, 0x6b, 0x14, 0x89, 0x15, 0x0c, 0x21, 0x51, 0x4e, 0xbf, 0x44,
     0x0f, 0xf5, 0xde, 0xa5, 0x39, 0x3d, 0x83, 0xde, 0x53, 0x58, 0xcd, 0x09, 0x8f, 0xce,
     0x8f, 0xd0, 0xf8, 0x1d, 0xaa, 0x94, 0x97, 0x91, 0x83
+#else
+#ifdef HAVE_CAL_TESTING_KEY_CI
```
[the remark this answers is not in the document]
Comment (apaillier-ledger) [reply src/tokens.h]: Indeed, should be better now :+1:

File: Makefile (hunk 2 above)
Comment (apaillier-ledger) [reply Makefile]: Indeed, should be better now :+1:

File: tests/ragger/conftest.py (hunk 3 above)
Comment (apaillier-ledger) [reply tests/ragger/conftest.py]: Indeed, should be better now :+1:

**Review by fbeutin-ledger — APPROVED** [review]

**Pull request merged**